File: VB/DXWebApplication1/Controllers/HomeController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Namespace DXWebApplication1.Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"

            Return View()
        End Function

        Private report As New DXWebApplication1.Reports.XtraReport1()

        Public Function DocumentViewerPartial() As ActionResult
            Return PartialView("_DocumentViewerPartial", report)
        End Function

        Public Function DocumentViewerPartialExport() As ActionResult
            Return DocumentViewerExtension.ExportTo(report, Request)
        End Function
    End Class
End Namespace

File: CS/DXWebApplication1/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using DevExpress.Web.Mvc;

namespace DXWebApplication1.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";

            return View();
        }

        private DXWebApplication1.Reports.XtraReport1 report = new DXWebApplication1.Reports.XtraReport1();

        public ActionResult DocumentViewerPartial() {
            return PartialView("_DocumentViewerPartial", report);
        }

        public ActionResult DocumentViewerPartialExport() {
            return DocumentViewerExtension.ExportTo(report, Request);
        }
    }
}
